pkg/advisory/request.go

```diff
@@ -35,8 +36,8 @@ func (req Request) Validate() error {
 		return err
 	}
 
-	if req.VulnerabilityID == "" {
-		return errors.New("vulnerability cannot be empty")
+	if req.VulnerabilityID != "" {
+		return errors.New("vulnerability should be empty")
```

Comment:
This check makes sense to me — but it also almost makes we want to remove the `vulnerabilityID` field from `Request` altogether, if it's only ever a footgun. Curious what you think here!

Reply:
i have no strong opinions here, but let me give a try to remove that, but i think will require more time to refactor the code

the other changes are in place, hope this one is not a blocker for you

Reply:
No not a blocker! Good call